internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -95,6 +95,15 @@ func (h *Host) SetSecurityPolicy(base64Policy string) error {
 		return err
 	}
 
+	hostData, err := securitypolicy.NewSecurityPolicyDigest(base64Policy)
+	if err != nil {
+		return err
+	}
+
+	if err := validateHostData(fmt.Sprintf("%x", hostData[:])); err != nil {
```

Comment:
Maybe I'm missing something, but shouldn't we be doing base64 encoding here instead of just `%x` formatting? It looks like the data we get from SEV is base64 encoded since that's what we pass in `create_lcow.go`.

Reply:
That's terrible lol. So HCS is decoding the base64 and formatting it as a hex string? If that's the case let's make a comment both here and in `create_lcow.go` since that is _super confusing_.

Reply:
sorry, should've been clearer here. I think it decodes the base64 encoded string into a byte array and SEV also returns bytes for HostData, I just encode it to string for better human readability. now that I think about it, maybe it's better to just do `bytes.Compare` instead?

Reply:
yeah bytes. No reason to encode just to compare.